### PR TITLE
BaSP-TG-69: Employees useForm

### PR DESCRIPTION
**Pull request by Baldizan**
Implemented useForm hook into employees' form.

### CODE REVIEW

File: src/Components/Employees/Form/index.js [hunk 1]
```diff
@@ -64,82 +64,93 @@ const EmployeesForm = () => {
...
-      {isPending && <p>Loading...</p>}
+      {isPending && <Loader />}
       {modalDisplay && (
         <Modal
           heading={
             error
               ? error
-              : `Employee ${employeeInput.name} ${employeeInput.lastName} submited successfully!`
+              : `Employee ${
+                  selectedEmployee ? `${selectedEmployee.name} ${selectedEmployee.lastName}` : null
```
Comment (GuidoG95) [inline src/Components/Employees/Form/index.js:152]: Instead of null you could show added employee name and last name

**Review by elias-92 — COMMENTED** [review]
Employee/index line 92, you should add a cancel button in the modal.

File: src/Components/Employees/Form/index.js [hunk 2]
```diff
@@ -1,54 +1,57 @@
...
-  );
+  const [feedback, setFeedback] = useState({ name: '', lastName: '' });
 
+  const {
+    handleSubmit,
+    register,
+    reset,
+    formState: { errors, isValid }
+  } = useForm({
+    mode: 'onChange',
```
Comment (GuidoG95) [inline src/Components/Employees/Form/index.js:30]: here we could use 'all' method so it still validates the field if onBlur as well as onChange

File: src/Components/Employees/Form/validations.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import Joi from 'joi';
+
+export const schema = Joi.object({
+  name: Joi.string().min(3).max(40).messages({
+    'string.min': 'First name must contain at least 3 letters.',
+    'string.max': 'First name cannot have over 40 letters.'
```
Comment (GuidoG95) [inline src/Components/Employees/Form/validations.js:6]: in all validation messages you could use 'string.empty' to show a custom message when the field is empty

**Review by LucianaPrat — APPROVED** [review]

**Review by elias-92 — APPROVED** [review]

**Review by mr-valen — APPROVED** [review]
Everything seems fine.

**Comment by mr-valen** [comment]
When submitting an employee the loader shows below the form instead of replacing it.

**Pull request merged**